File: course-info/simplify-json.py
import json
import sys


if __name__ == "__main__":
    simplified_course_list = [
        dict(
            courseId=course_json['crseId'],
            subject=course_json['subject'],
            courseNumber=course_json['catalogNbr'],
            title=course_json['titleShort']
        ) for course_json in json.load(sys.stdin)
    ]
    json.dump(simplified_course_list, sys.stdout, indent=2, sort_keys=True)

File: course-info/src/simplify-json.ts
import { readFile } from 'fs';
import { CourseInfo } from './types';

function main(): void {
  readFile(0 /* stdin */, 'utf8', (err, data) => {
    if (err) { throw err; }
    const complexCourseList = JSON.parse(data);
    const simplifiedCourseList = complexCourseList.map(
      ({ crseId: courseId, subject, catalogNbr: courseNumber, titleShort: title }): CourseInfo => ({
        courseId,
        subject,
        courseNumber,
        title,
      }),
    );
    // Need console.log to output to stdout.
    // eslint-disable-next-line no-console
    console.log(JSON.stringify(simplifiedCourseList));
  });
}

main();
